packages/block-library/src/social-link/index.php

```diff
@@ -45,6 +45,15 @@ function render_block_core_social_link( $attributes, $content, $block ) {
 		)
 	);
 
+	// Add default https if no protocol specified. The following chunk of code
+	// is taken from esc_url which is hard-coded to use http. This can be
+	// changed to pass the parameter in esc_url after
+	// https://core.trac.wordpress.org/ticket/52886 is in core.
+	if ( strpos( $url, ':' ) === false && ! in_array( $url[0], array( '/', '#', '?' ), true ) &&
+		! preg_match( '/^[a-z0-9-]+?\.php/i', $url ) ) {
```

Comment:
Could we add a few unit tests for this? For example, why would we be looking for a `.php` filename? Are there any other valid protocol-less links that folks would normally enter? 

To make this easier to test, we can pull this logic out to a helper function. See [example](https://github.com/WordPress/gutenberg/blob/trunk/phpunit/class-block-library-navigation-link-test.php ) in https://developer.wordpress.org/block-editor/contributors/code/testing-overview/#php-testing